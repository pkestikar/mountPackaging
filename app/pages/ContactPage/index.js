import React, { useState } from 'react';
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import PageHeader from "Components/PageHeader";
import { CONTACT_EMAIL } from "../../config";
import MoreInfoComposer from "Components/MoreInfoComposer";
import { useHistory } from 'react-router-dom';
import InputComposer from "Components/InputComposer";
import Button from 'Components/Button';

const ContactPage = (props) => {
    let history = useHistory();

    const [contactFormData, setContactFormData] = useState({
        name: {
            value: '',
            hasError: false,
            required: true
        },
        company: {
            value: '',
            hasError: false,
            required: true
        },
        email: {
            value: '',
            hasError: false,
            required: true
        },
        message: {
            value: '',
            hasError: false,
            required: false
        }
    });

    const messageCharLimit = 2000;

    const handleQuotationClick = () => {
        history.push('/contact/');
    };

    const handleInputChange = e => {
        let name = e.target.name;
        let newData = Object.assign({}, contactFormData);
        newData[name].value = e.target.value;
        if (name === 'message' && e.target.value.length > messageCharLimit) {
            newData[name].value = e.target.value.substring(0, messageCharLimit);
        }

        setContactFormData(Object.assign({}, contactFormData, newData));
    }

    const handleSendMessageClick = () => {
        let formComplete = true;
        let formData = Object.assign({}, contactFormData);

        for (const [key, value] of Object.entries(contactFormData)) {
            if (contactFormData[key].required && contactFormData[key].value.length < 1) {
                formComplete = false;
                formData[key].hasError = true;
            } else {
                formData[key].hasError = false;
            }
        }

        setContactFormData(Object.assign({}, contactFormData, formData));

        if (formComplete) {
            console.log('need to send this payload');
            console.log(contactFormData);
        }
    };

    return (
        <div className="mp--contact-page-container">
            <Container maxWidth={'lg'}>
                <PageHeader title={'Contact'}/>

                <Grid container justify={'center'} alignItems={'center'}>
                    <Grid item lg={4} className={'company-address-wrapper'}>
                        <span>We are located at:</span>
                        <p>
                            Mahatma Jyotirao Phule Marg, <br/>
                            Dhadge Industrial Estate, Nanded Phata, <br/>
                            Sinhgad Road, <br/>
                            Pune 411041, <br/>
                            Maharashtra, India. <br/>
                        </p>
                        <span className={'email'}>Email: <a href={`mailto: ${CONTACT_EMAIL}`}>{CONTACT_EMAIL}</a></span>
                    </Grid>
                </Grid>

                <h2 className={'contact-form-title'}>Let's Connect!</h2>

                <Grid container className={'contact-form-container'} justify={'center'} alignItems={'center'}>
                    <Grid item xs={12} sm={10} md={8}>
                        <form className={'mp--form-theme'}>
                            <InputComposer label={'Name'} hasError={contactFormData.name.hasError}>
                                <input type="text" name={'name'} id={'name'} value={contactFormData.name.value} onChange={e => handleInputChange(e)}/>
                            </InputComposer>
                            <InputComposer label={'Company'} hasError={contactFormData.company.hasError}>
                                <input type="text" name={'company'} id={'company'} value={contactFormData.company.value} onChange={e => handleInputChange(e)}/>
                            </InputComposer>
                            <InputComposer label={'E-Mail'} hasError={contactFormData.email.hasError}>
                                <input type="email" name={'email'} id={'email'} value={contactFormData.email.value} onChange={e => handleInputChange(e)}/>
                            </InputComposer>
                            <InputComposer label={`Message (Optional)`} hasError={contactFormData.message.hasError}>
                                <textarea rows={'6'} name={'message'} id={'message'} value={contactFormData.message.value} onChange={e => handleInputChange(e)}/>
                                <span className="text-limit-counter">
                                    {contactFormData.message.value.length} / {messageCharLimit}
                                </span>
                            </InputComposer>
                        </form>
                    </Grid>
                    <Grid item xs={12} className={'send-message-button-wrapper'}>
                        <Button onClick={handleSendMessageClick}>Send Message</Button>
                    </Grid>
                </Grid>
            </Container>
            <MoreInfoComposer
                title={'Ready with requirements?'}
                ctaText={'Get Quotation'}
                handleClick={handleQuotationClick}
                borderTop
            />
        </div>
    )
};

export default ContactPage;
