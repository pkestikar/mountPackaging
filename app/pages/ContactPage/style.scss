.mp--contact-page-container {
  margin-top: $page-container-top-margin__large;
  @include breakpoint(small down) {
    margin-top: $page-container-top-margin__small;
  }

  .company-address-wrapper {
    font-size: 1.1rem;
    margin-bottom: 2rem;
  }

  .contact-form-title {
    text-align: center;
    margin-bottom: 1rem;
  }

  .send-message-button-wrapper {
    width: 100%;
    text-align: center;
    margin-bottom: 1rem;
    button {
      margin: 1.5rem auto;
    }
  }
}