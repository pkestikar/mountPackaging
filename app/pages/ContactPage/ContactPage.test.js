import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import ContactPage from "Pages/ContactPage/index";

describe('ContactPage component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<ContactPage />);
    });

    it('renders correctly', () => {
         expect(toJSON(wrapper)).toMatchSnapshot();
    });


});
