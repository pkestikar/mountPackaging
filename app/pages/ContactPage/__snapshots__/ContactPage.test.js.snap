// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage component renders correctly 1`] = `
<div
  className="mp--contact-page-container"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
  >
    <PageHeader
      title="Contact"
    />
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        className="company-address-wrapper"
        item={true}
        lg={4}
      >
        <span>
          We are located at:
        </span>
        <p>
          Mahatma Jyotirao Phule Marg, 
          <br />
          Dhadge Industrial Estate, Nanded Phata, 
          <br />
          Sinhgad Road, 
          <br />
          Pune 411041, 
          <br />
          Maharashtra, India. 
          <br />
        </p>
        <span
          className="email"
        >
          Email: 
          <a
            href="mailto: mountindia@mountindia.net"
          >
            mountindia@mountindia.net
          </a>
        </span>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <h2
      className="contact-form-title"
    >
      Let's Connect!
    </h2>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="contact-form-container"
      container={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={8}
        sm={10}
        xs={12}
      >
        <form
          className="mp--form-theme"
        >
          <InputComposer
            label="NAME"
          >
            <input
              id="name"
              name="name"
              onChange={[Function]}
              type="text"
              value=""
            />
          </InputComposer>
          <InputComposer
            label="COMPANY"
          >
            <input
              id="company"
              name="company"
              onChange={[Function]}
              type="text"
              value=""
            />
          </InputComposer>
          <InputComposer
            label="EMAIL"
          >
            <input
              id="email"
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </InputComposer>
          <InputComposer
            label="MESSAGE"
          >
            <textarea
              id="message"
              name="message"
              onChange={[Function]}
              rows="6"
              value=""
            />
          </InputComposer>
        </form>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="send-message-button-wrapper"
        item={true}
        xs={12}
      >
        <Button
          disabled={false}
          loading={false}
          onClick={[Function]}
          palette="primary"
          size="medium"
        >
          Send Message
        </Button>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
  <MoreInfoComposer
    borderTop={true}
    ctaText="Get Quotation"
    handleClick={[Function]}
    normalFontSize={false}
    title="Ready with requirements?"
  />
</div>
`;
